File: data/derived_cot/rq1_traces/translation/qwen/java_to_cpp/code_output/FitnessTracker.cpp
#include <string>
#include <vector>

class FitnessTracker {
private:
    double height;
    double weight;
    int age;
    std::string sex;
    std::vector<std::vector<double>> BMI_std = {{20, 25}, {19, 24}};

public:
    explicit FitnessTracker(double height, double weight, int age, std::string sex)
        : height(height), weight(weight), age(age), sex(std::move(sex)) {}

    double getBMI() const {
        return weight / (height * height);
    }

    int conditionJudge() const {
        const int index = (sex == "male") ? 0 : 1;
        const double BMI = getBMI();
        if (BMI > BMI_std[index][1]) {
            return 1;
        }
        else if (BMI < BMI_std[index][0]) {
            return -1;
        }
        return 0;
    }

    double calculateCalorieIntake() const {
        const double BMR = (sex == "male") 
            ? (10 * weight + 6.25 * height - 5 * age + 5)
            : (10 * weight + 6.25 * height - 5 * age - 161);
        const int condition = conditionJudge();
        switch (condition) {
            case 1: return BMR * 1.2;
            case -1: return BMR * 1.6;
            default: return BMR * 1.4;
        }
    }
};

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/FitnessTracker.py
class FitnessTracker:

    def __init__(self, height, weight, age, sex):
        self.height = height
        self.weight = weight
        self.age = age
        self.sex = sex
        self.BMI_std = [[20.0, 25.0], [19.0, 24.0]]

    def get_BMI(self):
        return self.weight / (self.height * self.height)

    def condition_judge(self):
        BMI = self.get_BMI()
        index = 0 if self.sex == 'male' else 1
        if BMI > self.BMI_std[index][1]:
            return 1
        elif BMI < self.BMI_std[index][0]:
            return -1
        else:
            return 0

    def calculate_calorie_intake(self):
        if self.sex == 'male':
            BMR = 10 * self.weight + 6.25 * self.height - 5 * self.age + 5
        else:
            BMR = 10 * self.weight + 6.25 * self.height - 5 * self.age - 161
        condition = self.condition_judge()
        if condition == 1:
            return BMR * 1.2
        elif condition == -1:
            return BMR * 1.6
        else:
            return BMR * 1.4